ext/standard/tests/network/bug10765.phpt

```diff
@@ -0,0 +1,10 @@
+--TEST--
+setcookie() expiry time negative warning
+--FILE--
+<?php
+setcookie('name', 'value', -1);
+?>
+--EXPECTF--
+Warning: setcookie(): "expires" option cannot be negative in %s on line %d
+
+Warning: Cannot modify header information - headers already sent by (output started at %s:%d) in %s on line %d
```

Comment:
This is unfortunate side effect, without output buffering cookie will never be sent..

Also seems like `-1` is used in some code, and this will be probably a breaking change for them https://grep.app/search?q=setcookie%5C%28%5B%5E%2C%5D%2A%2C%5B%5E%2C%5D%2A%2C%5Cs%2A-1%5B%5E%5Cd%5D&regexp=true&filter[lang][0]=PHP

Reply:
Agree, with this we have not much to do here.

From what I understand looking at your search, -1 is used to invalidate a cookie.

I thought of some changes, but all will have a compatibility break (from what I see).

- We can throw an argument error when reaching `-1`.

- We can validate [here](https://github.com/php/php-src/blob/aef5250eae2bceba4300cba986c33504778d0f88/ext/standard/head.c#L123) the expires and invalidate the cookie if it comes negative.